### PR TITLE
gh-102541: Fix Helper.help("mod") for non-existent mod

### PR DESCRIPTION
**Pull request by Eclips4**
If the `output` arg to `Helper()`  is a stream rather than the default None, which means 'page to stdout', the ImportError from `pydoc.resolve` is currently not caught in `pydoc.doc.  The same error *is* caught when output is None.
```python3
import pydoc
from io import StringIO

input = StringIO()
output = StringIO()
help = pydoc.Helper(input, output)
help.help("abd")
input.close()
output.close()
```
Traceback:
```python3
./python example.py
Running debug|x64 interpreter...
Traceback (most recent call last):
  File "C:\Users\KIRILL-1\CLionProjects\cpython\example.py", line 7, in <module>
    help.help("abd", is_cli=True)
  File "C:\Users\KIRILL-1\CLionProjects\cpython\Lib\pydoc.py", line 2067, in help
    elif request: doc(request, 'Help on %s:', output=self._output, is_cli=is_cli)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\KIRILL-1\CLionProjects\cpython\Lib\pydoc.py", line 1793, in doc
    output.write(render_doc(thing, title, forceload, plaintext))
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\KIRILL-1\CLionProjects\cpython\Lib\pydoc.py", line 1760, in render_doc
    object, name = resolve(thing, forceload)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\KIRILL-1\CLionProjects\cpython\Lib\pydoc.py", line 1746, in resolve
    raise ImportError('''\
ImportError: No Python documentation found for 'abd'.
Use help() to get the interactive help utility.
Use help(str) for help on the str class.
```
EDITED by terryjreedy July 1, 2023 to better describe the issue, so I could review the patch,
and show the current result after the merge of the first PR for the issue.  Instead of the last line
there were about 15 lines that repeated the Traceback.

* Issue: gh-102541

### CODE REVIEW

**Comment by corona10** [comment]
@terryjreedy @Yhg1s @ambv 

I am out of OSS for this month. If this PR is urgent, I would like to request reviewing of this PR.

File: Lib/test/test_pydoc.py [hunk 1]
```diff
@@ -631,6 +631,14 @@ def test_builtin_on_metaclasses(self):
         # Testing that the subclasses section does not appear
         self.assertNotIn('Built-in subclasses', text)
 
+    def test_fail_help_output_redirect(self):
+        buf = StringIO()
```
Comment (sobolevn) [inline Lib/test/test_pydoc.py]: nit: why not `with StringIO() as buf:`?

Comment (Eclips4) [reply Lib/test/test_pydoc.py]: > nit: why not `with StringIO() as buf:`?

Good point. Done

File: Lib/pydoc.py [hunk 2]
```diff
@@ -1790,7 +1790,10 @@ def doc(thing, title='Python Library Documentation: %s', forceload=0,
                 raise
             print(exc)
     else:
-        output.write(render_doc(thing, title, forceload, plaintext))
+        try:
+            output.write(render_doc(thing, title, forceload, plaintext))
+        except ImportError as exc:
+            output.write(str(exc))
```
Comment (terryjreedy) [inline Lib/pydoc.py]: A general principle is to put at little in the try suite as needed to get the exception.
```suggestion
        try:
            s = render_doc(thing, title, forceload, plaintext))
        except ImportError as exc:
            s = str(exc))
        output.write(s)
```
I am going to merge with this change.

**Pull request merged**